bulk_indexer.go

```diff
@@ -132,8 +137,10 @@ func init() {
 	})
 }
 
-func newBulkIndexer(client *elasticsearch.Client, compressionLevel int) *bulkIndexer {
+func newBulkIndexer(client *elasticsearch.Client, compressionLevel int, maxRetryDoc int) *bulkIndexer {
 	b := &bulkIndexer{client: client}
+	b.maxDocumentRetry = maxRetryDoc
+	b.retryCounts = make(map[int]int)
```

Comment:
I think it looks a bit more consistent and concise this way. WDYT?
```suggestion
func newBulkIndexer(client *elasticsearch.Client, compressionLevel int, maxDocRetry int) *bulkIndexer {
	b := &bulkIndexer{
		client:           client,
		maxDocumentRetry: maxDocRetry,
		retryCounts:      make(map[int]int),
	}
```